### PR TITLE
Add ESA Worldcover indexing

### PR DESCRIPTION
**Pull request by alexgleith**
* Adds ESA Worldcover indexing tool
* Fixes naming of tests for the copernicus DEM indexing

### CODE REVIEW

**Review by carioca-au — APPROVED** [review]

File: apps/dc_tools/odc/apps/dc_tools/esa_worldcover_to_dc.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+import rasterio
+from datacube import Datacube
+from datacube.index.hl import Doc2Dataset
+from datacube.utils import read_documents
+from odc.apps.dc_tools.utils import bbox, index_update_dataset, limit, update_if_exists
+from rio_stac import create_stac_item
+
+from ._stac import stac_transform
+
+PRODUCT = (
```
Comment (carioca-au) [inline apps/dc_tools/odc/apps/dc_tools/esa_worldcover_to_dc.py:22]: if you have a function to add the product and a flag to see if the product has to be added, shouldn't you send the path when you want to add or update the product instead have it hardcoded?

File: apps/dc_tools/odc/apps/dc_tools/esa_worldcover_to_dc.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    bounding_box,
+    limit: int,
+    update: bool,
+    n_workers: int = 100,
+) -> Tuple[int, int]:
+    doc2ds = Doc2Dataset(dc.index)
+
+    # Get a generator of (uris)
+    uris_tiles = list(get_tile_uris(bounding_box))
+    if limit:
```
Comment (carioca-au) [inline apps/dc_tools/odc/apps/dc_tools/esa_worldcover_to_dc.py:148]: Would be better to convert to a list just if you have a limit. There is no need to use a list if you will just loop over that

**Review by carioca-au — APPROVED** [review]
Just 2 minor sugestions

File: apps/dc_tools/odc/apps/dc_tools/esa_worldcover_to_dc.py (hunk 2 above)
Comment (alexgleith) [reply apps/dc_tools/odc/apps/dc_tools/esa_worldcover_to_dc.py:148]: Could be.

File: apps/dc_tools/odc/apps/dc_tools/esa_worldcover_to_dc.py (hunk 1 above)
Comment (alexgleith) [reply apps/dc_tools/odc/apps/dc_tools/esa_worldcover_to_dc.py:22]: there's only one product

**Pull request merged**